bastion/src/context.rs

```diff
@@ -346,22 +347,30 @@ impl BastionContext {
     ///
     /// [`try_recv`]: #method.try_recv
     /// [`Msg`]: children/struct.Msg.html
-    pub async fn recv(&self) -> Result<Msg, ()> {
+    pub async fn recv(&self) -> Result<SignedMessage, ()> {
         debug!("BastionContext({}): Waiting to receive message.", self.id);
         loop {
             // TODO: Err(Error)
             let mut state = self.state.clone().lock_async().await.unwrap();
 
-            if let Some(msg) = state.msgs.pop_front() {
+            if let Some((msg, sign)) = state.msgs.pop_front() {
                 trace!("BastionContext({}): Received message: {:?}", self.id, msg);
-                return Ok(msg);
+                return Ok((msg, sign));
             }
 
             Guard::unlock(state);
 
             pending!();
         }
     }
+
+    /// Returns [`Signature`] if the current `BastionContext`
```

Comment:
It would be nice to have an example like other methods.

Reply:
fixed